src/main/kotlin/app/revanced/patches/youtube/misc/videoid/patch/VideoIdPatch.kt

```diff
@@ -0,0 +1,56 @@
+package app.revanced.patches.youtube.misc.videoid.patch
+
+import app.revanced.patcher.annotation.Description
+import app.revanced.patcher.annotation.Name
+import app.revanced.patcher.annotation.Version
+import app.revanced.patcher.data.impl.BytecodeData
+import app.revanced.patcher.extensions.addInstructions
+import app.revanced.patcher.patch.PatchResult
+import app.revanced.patcher.patch.PatchResultSuccess
+import app.revanced.patcher.patch.annotations.Dependencies
+import app.revanced.patcher.patch.impl.BytecodePatch
+import app.revanced.patches.youtube.misc.integrations.patch.IntegrationsPatch
+import app.revanced.patches.youtube.misc.videoid.annotation.VideoIdCompatibility
+import app.revanced.patches.youtube.misc.videoid.fingerprint.VideoIdFingerprint
+import org.jf.dexlib2.iface.instruction.formats.Instruction11x
+
+@Name("video-id-hook")
+@Description("hook to detect when the video id changes")
+@VideoIdCompatibility
+@Version("0.0.1")
+@Dependencies(dependencies = [IntegrationsPatch::class])
+class VideoIdPatch : BytecodePatch(
+    listOf(
+        VideoIdFingerprint
+    )
+) {
+    override fun execute(data: BytecodeData): PatchResult {
+        injectCall("Lapp/revanced/integrations/videoplayer/VideoInformation;->setCurrentVideoId(Ljava/lang/String;)V")
+
+        return PatchResultSuccess()
+    }
+
+    companion object {
+        private var offset = 2
+
+        /**
+         * Adds an invoke-static instruction, called with the new id when the video changes
+         * @param methodDescriptor which method to call. Params have to be `Ljava/lang/String;`
+         */
+        fun injectCall(
```

Comment:
Is this meant to be public? For the scope of this patch, this should be private.

Reply:
Yes, that way patches that also need it can add their own methods like:
https://github.com/revanced/revanced-patches/pull/175/files#diff-3897157b7579c9e0f9618a8b8fe3c1e0bc754afb7aba449fd7d96aa9ff8cae8bR56